### PR TITLE
[Merged by Bors] - refactor: generalize shifts from bitvector to arbitrary vectors

### PR DESCRIPTION
**Pull request by alexkeizer**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Vector.lean [hunk 1]
```diff
@@ -13,6 +13,7 @@ import Std.Data.List.Basic
 import Std.Data.List.Lemmas
 import Mathlib.Init.Data.List.Basic
 import Mathlib.Init.Data.List.Lemmas
+import Mathlib.Data.Nat.Pow
```
Comment (eric-wieser) [inline Mathlib/Data/Vector.lean:16]: Is this actually needed?

Comment (alexkeizer) [reply Mathlib/Data/Vector.lean:16]: I don't think definitions from `Data.Nat.Pow` itself are needed, but it does transitively import a bunch of files that are needed, and I couldn't find which files these are.
Specifically, I couldn't find where an instance for `ExistsAddOfLE ℕ` is defined, but importing that file seems to give it.

File: Mathlib/Data/Bitvec/Defs.lean [hunk 2]
```diff
@@ -69,33 +69,18 @@ variable {n : ℕ}
...
-  Bitvec.cong
-      (by
-        by_cases h : i ≤ n
-        · have h₁ := Nat.sub_le n i
-          rw [min_eq_right h]
-          rw [min_eq_left h₁, ← add_tsub_assoc_of_le h, Nat.add_comm, add_tsub_cancel_right]
-        · have h₁ := le_of_not_ge h
-          rw [min_eq_left h₁, tsub_eq_zero_iff_le.mpr h₁, zero_min, Nat.add_zero]) <|
-    replicate (min n i) fill++ₜtake (n - i) x
-#align bitvec.fill_shr Bitvec.fillShr
```
Comment (eric-wieser) [inline Mathlib/Data/Bitvec/Defs.lean:88]: This should either be no-aligned, or the PR should be labelled `after-port`.

Comment (alexkeizer) [reply Mathlib/Data/Bitvec/Defs.lean:88]: I did a quick search in mathlib(3), and didn't find any downstream uses of `bitvec.fill_shr`, so I no-aligned it

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Pull request closed without merging**